### PR TITLE
Add ITEMS_COUNT and ITEM_NAME_LENGTH to RHH rom header

### PR DESCRIPTION
**Pull request by Ninjdai1**
## Description
There was no way for an external program to access the ITEMS_COUNT or ITEM_NAME_LENGTH.
This PR adds ITEMS_COUNT and ITEM_NAME_LENGTH to the RHH rom header*

## **Discord contact info**
ninjdai

### CODE REVIEW

File: src/rom_header_rhh.c [hunk 1]
```diff
@@ -21,6 +22,8 @@ struct RHHRomHeader
     /*0x10*/ u32 numSpecies;
     /*0x14*/ u32 abilitiesCount;
     /*0x18*/ const struct Ability *abilities;
+    /*0x1C*/ u32 itemsCount;
+    /*0x20*/ u8 itemNameLength;
```
Comment (AsparagusEduardo) [inline src/rom_header_rhh.c:26]: To make the most out of this space, let's add some padding here for other `u8`s
```suggestion
    /*0x20*/ u8 itemNameLength;
    /*0x21*/ u8 padding[3];
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

File: src/rom_header_rhh.c [hunk 2]
```diff
@@ -21,6 +22,9 @@ struct RHHRomHeader
     /*0x0C*/ u16 numSpecies;
     /*0x0E*/ u16 abilitiesCount;
     /*0x10*/ const struct Ability *abilities;
+    /*0x14*/ u32 itemsCount;
```
Comment (GraionDilach) [inline src/rom_header_rhh.c:25]: Shouldn't this also be u16?

Comment (Ninjdai1) [reply src/rom_header_rhh.c:25]: It should, that's an oversight

File: src/rom_header_rhh.c [hunk 3]
```diff
@@ -21,6 +22,9 @@ struct RHHRomHeader
     /*0x0C*/ u16 numSpecies;
     /*0x0E*/ u16 abilitiesCount;
     /*0x10*/ const struct Ability *abilities;
+    /*0x14*/ u16 itemsCount;
+    /*0x16*/ u8 itemNameLength;
+    /*0x17*/ u8 padding[3];
```
Comment (AsparagusEduardo) [inline src/rom_header_rhh.c:27]: Now that `itemsCount` is u16, there's no need for the 2 extra bytes of padding.
```suggestion
    /*0x17*/ u8 padding;
```

**Review by Bassoonian — APPROVED** [review]

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**